### PR TITLE
`Development`: Update client dependencies to Angular 15 and Jest 29

### PR DESCRIPTION
**Pull request by krusche**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context
Artemis currently has Angular v.14. We want to update Angular client to the latest major v.15, as the major releases can include the following benefits for the evolution of the angular framework:
* Release of useful new features.
* Better performance.

Moreover, each major version support usually lasts 18 months, with 6 months active support phase and 12 months LTS. During this time, the critical issues are fixed, and security patches are released.

Artemis also uses Jest Testing Framework v.28, which we want to update to v. 29. 

### Description
* Using the official [upgrade guide](https://update.angular.io/?l=3&v=14.0-15.0) we upgrade the Angular v. 14 to v. 15.
* Using the official [upgrade guide](https://jestjs.io/docs/upgrading-to-jest29) we upgraded the Jest v. 28 to v. 29

### Steps for Testing

Prerequisites:
- 1 admin

Check the angular client, if everything works as expected after the update:
* Check all components as a user 
* Check all tests run: 
* Check if build is succeed using the following command in terminal:

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2
#### Exam Mode Test
- [x] Test

### CODE REVIEW

File: src/main/webapp/content/scss/global.scss [hunk 1]
```diff
@@ -6,9 +6,9 @@
 @import 'node_modules/bootstrap/scss/bootstrap';
 @import 'node_modules/@danielmoncada/angular-datetime-picker/assets/style/picker.min';
 // Needed for mathematical formulas in markdown editor
-@import '~katex/dist/katex.css';
-@import '~@flaviosantoro92/ngx-datatable/themes/bootstrap.css';
-// Needed for emojis in metis reactions
+@import 'node_modules/katex/dist/katex.min.css';
+@import 'node_modules/@flaviosantoro92/ngx-datatable/themes/bootstrap';
+// Needed for emojis in metis reactionsx
```
Comment (Strohgelaender) [inline src/main/webapp/content/scss/global.scss:11]:
```suggestion
// Needed for emojis in metis reactions
```

**Review by danielbode — APPROVED** [review]
Tested all views as a student and participated in a Quiz Exercise, Programming Exercise, Text Exercise, and Modeling Exercise on ts1 during a testing session. Everything worked fine.

**Review by evgeny1984 — APPROVED** [review]
Tested during testing session on the ts1 following UI components:
- Server administration
- User profile
- Notification
- Exam Mode: Creation, Import, Settings
Everything worked as expected.

**Review by TheZoker — APPROVED** [review]
Tested course management and exam participation in test session on TS1

**Review by rriyaldhi — DISMISSED** [review]
Tested on TS1 as an Instructor and as a Student in Exam mode.

**Review by stefanwaldhauser — DISMISSED** [review]
Tested ott locally and everything seemed tow work fine

**Review by danielbode — APPROVED** [review]
Re-tested after changes. Didn't notice any bugs, still looks good to me.

**Review by evgeny1984 — APPROVED** [review]
Re approved after small changes and tested on ts1. Everything works as exptected.

**Pull request merged**